go/vt/vtgate/buffer/buffer_test.go

```diff
@@ -72,6 +72,15 @@ func TestBuffering(t *testing.T) {
 	testAllImplementations(t, testBuffering1)
 }
 
+func TestBufferingConcurrent(t *testing.T) {
+	tmp := bufferDrainConcurrency
```

Comment:
We are not running tests with `t.Parallel()` so this _should_ be OK.

Reply:
We already have a `DrainConcurrency` inside `Config`. Shouldn't we pass in the config then and not depend in the buffer code on this global? But properly configure this? That seems like a much more desirable way to pass this in. 

We can set it from the config on the `shardBuffer` then instead.

Reply:
Yeah, we already have this config on the `Buffer`, we should pass it into the constructor for `shardBuffer` then and store it there and use it from there and not use a global. Makes it easier to test and less magic at a distance with globals.

Reply:
yeah this seemed bad, was hoping someone would show me the way :)